feat: CPLYTM-665 CPLYTM-664 Implementation for sync-oscal-content profile command

RichardXuan opened the pull request:
## Description

<!--Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.-->

Implementation for sync-oscal-content profile command, this cmd will sync OSCAL control level change to CaC side.

Fixes CPLYTM-665

```shell
poetry run trestlebot sync-oscal-content profile --dry-run --repo-path ~/trestlebot-workspace --committer-email test@redhat.com --committer-name test --branch main --cac-content-root ~/content --cac-policy-id cis_rhel8 --product rhel8 
```


## Type of change

<!--Please delete options that are not relevant.-->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested?

<!--Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration-->

- [x] Local manual test
- [x] Unit test

**Test Configuration**:

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

jpower432 commented on trestlebot/tasks/sync_osacl_content_profile_task.py:
Nit: minor typo in file name `osacl -> oscal`

jpower432 reviewed (commented):
Small nit in the file name, but everything looks good. I tested locally with the provided command and was able to confirm behavior was as expected.

RichardXuan commented:
@jpower432 Thanks. Jenn. Fixed file name typo.

huiwangredhat reviewed (commented):
If I removed the level from profiles, example: `
processing l1_workstation, add: set(), remove: set()
processing l1_server, add: set(), remove: set()
processing l2_server, add: set(), remove: set()
processing l2_workstation, add: set(), remove: {'cis_rhel8_1-1.1.7'}
Staging all repository changes`, 
The control file hasn't updated`                    
 - id: 1.1.1.7
    title: Ensure udf kernel module is not available (Automated)
    levels:
      - l2_server
      - l2_workstation` --> the expected result should remove this line, am I right?

RichardXuan commented:
> I have one question. If I removed the level from profiles, example: ` processing l1_workstation, add: set(), remove: {'cis_rhel8_1-1.2.1.1'}`, the control file will be updated like this ` - id: 1.1.2.1.1 title: Ensure /tmp is a separate partition (Automated) levels: - l1_server - l2_workstation`

@huiwangredhat  This is expected, check https://github.com/ComplianceAsCode/content/blob/master/controls/cis_rhel8.yml#L13 ,  `l2_workstation` inherits_from `l1_workstation` , this means `l2_workstation` is a higher level and including all controls from `l1_workstation` level . When control `1.1.2.1.1` removed from `l1_workstation` level, it still belong to `l2_workstation`.

huiwangredhat commented:
Another question, we need something like this --cac-banch to make sure to update to the right branch of the cac content repo.

RichardXuan commented:
> If I removed the level from profiles, example: ` processing l1_workstation, add: set(), remove: set()
> 
> processing l1_server, add: set(), remove: set()
> 
> processing l2_server, add: set(), remove: set()
> 
> processing l2_workstation, add: set(), remove: {'cis_rhel8_1-1.1.7'}
> 
> Staging all repository changes`,
> 
> The control file hasn't updated`
> 
> * id: 1.1.1.7
>   title: Ensure udf kernel module is not available (Automated)
>   levels:
>   
>   * l2_server
>   * l2_workstation` --> the expected result should remove this line, am I right?

@huiwangredhat Updated. Fixed this. Please try again. Thanks.

jpower432 commented:
> Another question, we need something like this --cac-banch to make sure to update to the right branch of the cac content repo.

@huiwangredhat You bring up a really good point here. If I review this from a CI workspace perspective, the workflow seems to work like this:

1. Clone repository with trestle-workspace
2. Clone the CaC repo
3. Run the `sync-oscal-content` command
4. The CaC content repo is content updated, but the I receive is `Nothing to commit` because TrestleBot is looking changes in the `trestle-workspace`.

@RichardXuan I would think that nothing should change in the `trestle-workspace` as a results of this command. Is that correct? Is the goal to have `trestle-bot` push to the `content` repo or just support transformation and use other GitHub Actions to commit and push?

I see that the CLI command was introduced this [PR](https://github.com/complytime/trestle-bot/pull/517). Please let me know if this is out of scope and will be handled in another PR.

RichardXuan commented:
> > Another question, we need something like this --cac-banch to make sure to update to the right branch of the cac content repo.
> 
> @huiwangredhat You bring up a really good point here. If I re-review this from a running in CI perspective, the workflow seems to work like this:
> 
> 1. Clone repository with trestle-workspace
> 2. Clone the CaC repo
> 3. Run the `sync-oscal-content` command
> 4. The CaC content repo is content updated, but then the message I receive is `Nothing to commit` because TrestleBot is looking for changes in the `trestle-workspace`.
> 
> I would think that nothing should change in the `trestle-workspace` as a result of this command so the outcome makes sense.
> 
> @RichardXuan Maybe there are some options we could explore to apply these updates to the CaC repo.
> 
> 1. `trestle-bot` pushes to the `content` repo. I think this could be done between passing in the cac-content repo as the working dir [here](https://github.com/complytime/trestle-bot/blob/90acf48208d5a2999a3ecdd990e4c10b606bd4c6/trestlebot/tasks/sync_oscal_content_profile_task.py#L31) and passing in the `trestle_root` as a argument to the task.
> 2. Another option might be to just support transformation to other formats as a general utility and use other GitHub Actions to commit and push. This might make sense if we see this transformation extending to different formats.
> 
> What do you think?
> 
> I see that the CLI command was introduced this [PR](https://github.com/complytime/trestle-bot/pull/517). Please let me know if my comment is out of scope and could be handled in another PR.

@jpower432 @huiwangredhat I think option 1 makes sense for now. WDYT. I will do this in this PR. Should be a simple change. This is something i forget to think about when adding `sync-oscal-content`, thanks for your remind.

huiwangredhat commented:
@RichardXuan I run it locally, it could remove the related levels:
Removed the control cis_rhel8_1-1.1.1 from oscal profile l1_workstation, it will remove the controls

```
processing l1_workstation, add: set(), remove: {'cis_rhel8_1-1.1.1'}
processing l1_server, add: set(), remove: set()
processing l2_server, add: set(), remove: set()
processing l2_workstation, add: set(), remove: {'cis_rhel8_1-1.1.1'}
```
Removed the control cis_rhel8_1-1.1.1 from oscal profile l2_workstation, it will remove the controls
```
processing l1_workstation, add: set(), remove: set()
processing l1_server, add: set(), remove: set()
processing l2_server, add: set(), remove: set()
processing l2_workstation, add: set(), remove: {'cis_rhel8_1-1.1.3'}
```

RichardXuan commented:
@huiwangredhat Now `sync-oscal-content` command performed git operation on CaC content repo. Please check it, thanks.

jpower432 reviewed (approved):
LGTM. Tested locally and the content local git repo was updated on the branch defined by `--branch`.

huiwangredhat reviewed (approved):
Thanks for the quick update. LGTM.
